### PR TITLE
Basic numeric type support : first cut

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Adds minimal support to `daml2ts` for DAML `Numeric n` values. Next step, write a smarter decoder for them than `jtv.string`.

changelog_begin
changelog_end

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -340,6 +343,7 @@ genType curModName = go
         TCon _ -> error "IMPOSSIBLE: lonely type constructor"
         TSynApp{} -> error "IMPOSSIBLE: type synonym not serializable"
         t@TApp{} -> error $ "IMPOSSIBLE: type application not serializable - " <> DA.Pretty.renderPretty t
+          -- The above case also handles 'TNumeric a' where 'a' is not a type level nat.
```
Comment (hurryabit) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: I don't think this comment really helps. Let's just drop it.

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts [hunk 2]
```diff
@@ -112,6 +112,26 @@ export const Decimal: Serializable<Decimal> = {
...
+ * in order to avoid a loss of precision. The string must match the regular
+ * expression `-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?`.
+ */
+export type Numeric = string;
+
+/**
+ * Companion function of the `Numeric` type.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const Numeric = function (_: number): Serializable<Numeric> {
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: That is a quite peculiar style for defining a function. We should either go for `export function Numeric(_: number): Serializable<Numeric> { ... }` or for `export const Numeric = (_: number): Serializable<Numeric> => ...`. To fit the style of the surrounding code, I suggest
```suggestion
export const Numeric = (_: number): Serializable<Numeric> => ({
  decoder: jtv.string,
});
```

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts [hunk 3]
```diff
@@ -112,6 +112,26 @@ export const Decimal: Serializable<Decimal> = {
...
+  // To-do : Write a smarter decoder to replace 'jtv.string' that
```
Comment (hurryabit) [inline language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: Can you please make it `//TODO(SF, #xyz): ...`, where `#xyz` is the number of the issue for this. I prefer to be consistent on how to leave todos. That makes it easier to grep for todos.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (shayne-fletcher) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: Remove comment

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts (hunk 2 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: Will fat arrow this.

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts (hunk 3 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: Ok.

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts [hunk 4]
```diff
@@ -112,6 +112,26 @@ export const Decimal: Serializable<Decimal> = {
   decoder: jtv.string,
```
Comment (shayne-fletcher) [inline language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: Rewrite this as an alias for `Numeric 10`.

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: ✔️

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts (hunk 4 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: ✔️

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts (hunk 2 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: ✔️

File: language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts (hunk 3 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/tests/ts/daml-json-types/src/index.ts]: ✔️

**Pull request merged**